### PR TITLE
Add tests for Identify functions and fix some issues identified along the way

### PR DESCRIPTION
**Pull request by calebsander**
Use the mock `ioctl()` infrastructure (#688) to test the functions in `ioctl.h`
that issue Identify commands.
`nvme_identify_ns_csi_user_data_format()` and
`nvme_identify_iocs_ns_csi_user_data_format()` are omitted
since they're not defined in the NVMe specification yet.
Functions tested indirectly from other functions are also omitted.

Fix a couple of bugs:
- `nvme_identify_secondary_ctrl_list()` shouldn't take a `nsid` parameter according to the NVMe specification
- The main `test` program is passing a buffer than is too short to `nvme_identify_ns_descs()`. Allocate a 4 KB buffer to avoid stack overflow.

And simplify a few of the Identify functions by delegating to available helper functions.

Also remove all the entries in `libnvme.map` that don't correspond to exported symbols.

### CODE REVIEW

**Review by igaw — CHANGES_REQUESTED** [review]
Wow, looks great. Just a small thing in the first patch.

File: src/libnvme.map [hunk 1]
```diff
@@ -3,6 +3,7 @@ LIBNVME_1_6 {
 	global:
 		nvme_ctrl_release_fd;
 		nvme_host_release_fds;
+		nvme_identify_secondary_ctrl_list;
```
[the remark this answers is not in the document]
Comment (calebsander) [reply src/libnvme.map:6]: Sounds good, I'll just drop it from the export list.
Curious what the `LIBNVME_1_X` sections in the version script are for if not to distinguish different versions of the functions? Are they purely to document which version added which function? Sorry, version scripts are new to me.

[the remark this answers is not in the document]
Comment (igaw) [reply src/libnvme.map:6]: I think we switched to the inline function later during the development phase and forgot to kill it in the map.

The different sections names are added to the function names. We try to make sure that we maintain backwards compatibility. Though I am also not totally sure it it works or not. When I looked into this I saw that other projects are doing this (like systemd). Looked reasonable at that time :)

[the remark this answers is not in the document]
Comment (calebsander) [reply src/libnvme.map]: Went ahead and removed all the nonexistent symbols from `libnvme.map`

[the remark this answers is not in the document]
Comment (calebsander) [reply src/libnvme.map]: Went ahead and removed all the nonexistent symbols from `libnvme.map`

**Review by igaw — APPROVED** [review]

**Pull request merged**